provider/lxd/environ.go

```diff
@@ -104,11 +105,36 @@ func (env *environ) profileName() string {
 	return "juju-" + env.Name()
 }
 
+type LXDProfiler interface {
+	MaybeWriteLXDProfile(pName string, put *charm.LXDProfile) error
+	Name() string
+}
+
 // Name returns the name of the environ.
 func (env *environ) Name() string {
 	return env.name
 }
 
+// MaybeWriteLXDProfile, write given LXDProfiler to if not already there.
+func (env *environ) MaybeWriteLXDProfile(pName string, put *charm.LXDProfile) error {
+	hasProfile, err := env.server.HasProfile(pName)
+	if err != nil {
+		return errors.Trace(err)
+	}
+	if hasProfile {
```

Comment:
Can we guarantee that this is the same Profile that we're putting up, does it harm anything if we push this again, but maybe log out (trace/info level) that the profile already exists?

Reply:
LXD errors out if you attempt to create a profile that already exists, logging a message if hasProfile is true.